src/mongo/delegates/Views.class.php

```diff
@@ -1,54 +1,120 @@
+
+/**
+ * Class Views
+ * @package Tripod\Mongo\Composites
+ */
+class Views extends CompositeBase
 {
 
     /**
      * Construct accepts actual objects rather than strings as this class is a delegate of
-     * MongoTripod and should inherit connections set up there
+     * Tripod and should inherit connections set up there
      * @param string $storeName
-     * @param MongoCollection $collection
+     * @param \MongoCollection $collection
      * @param $defaultContext
      * @param null $stat
      */
-    function __construct($storeName, MongoCollection $collection,$defaultContext,$stat=null)
+    function __construct($storeName, \MongoCollection $collection,$defaultContext,$stat=null) // todo: $collection -> podname
     {
         $this->storeName = $storeName;
-        $this->labeller = new MongoTripodLabeller();
+        $this->labeller = new Labeller();
         $this->collection = $collection;
         $this->podName = $collection->getName();
         $this->defaultContext = $defaultContext;
-        $this->config = MongoTripodConfig::getInstance();
+        $this->config = Config::getInstance();
         $this->stat = $stat;
+        $this->readPreference = \MongoClient::RP_PRIMARY; // todo: figure out where this should go.
+    }
+
+    /**
+     * @return string
+     */
+    public function getOperationType()
+    {
+        return OP_VIEWS;
     }
 
     /**
-     * (PHP 5 &gt;= 5.1.0)<br/>
      * Receive update from subject
-     * @link http://php.net/manual/en/splobserver.update.php
-     * @param SplSubject $subject <p>
-     * The <b>SplSubject</b> notifying the observer of an update.
-     * </p>
+     * @param \Tripod\Mongo\ImpactedSubject
      * @return void
      */
-    public function update(SplSubject $subject)
+    public function update(\Tripod\Mongo\ImpactedSubject $subject)
     {
-        /* @var $subject ModifiedSubject */
-        $queuedItem = $subject->getData();
-        $resourceUri    = $queuedItem[_ID_RESOURCE];
-        $context        = $queuedItem[_ID_CONTEXT];
+        $resource = $subject->getResourceId();
+        $resourceUri    = $resource[_ID_RESOURCE];
+        $context        = $resource[_ID_CONTEXT];
 
         $this->generateViews(array($resourceUri),$context);
     }
 
+    /**
+     * @return array
+     */
+    public function getTypesInSpecifications()
+    {
+        return $this->config->getTypesInViewSpecifications($this->storeName, $this->getPodName());
+    }
+
+    /**
+     * @param array $resourcesAndPredicates
+     * @param string $contextAlias
+     * @return array|mixed
+     */
+    public function findImpactedComposites(Array $resourcesAndPredicates, $contextAlias)
+    {
+        $resources = array_keys($resourcesAndPredicates);
+
+        $contextAlias = $this->getContextAlias($contextAlias); // belt and braces
+
+        // build a filter - will be used for impactIndex detection and finding direct views to re-gen
+        $filter = array();
```

Comment:
I noticed further up that you had in a similar function - `if(empty($query)) return array()` (paraphrased a bit).. do you want something similar here? i.e. if `$resources` is empty, can you just return an empty array?